src/libm/sleefdp.c

```diff
@@ -849,20 +859,27 @@ EXPORT CONST double xcos(double d) {
 EXPORT CONST double xcos_u1(double d) {
   double u;
   Sleef_double2 s, t, x;
-
+  int ql;
+  
   d = fabsk(d);
 
-  double dqh = trunck(d * (M_1_PI / (1LL << 23)) - 0.5 * (M_1_PI / (1LL << 23)));
-  int ql = 2*rintk(d * M_1_PI - 0.5 - dqh * (double)(1LL << 23))+1;
-  dqh *= 1 << 24;
-
-  u = mla(dqh, -PI_A*0.5, d);
-  s = ddadd2_d2_d_d (u,  ql * (-PI_A*0.5));
-  s = ddadd2_d2_d2_d(s, dqh * (-PI_B*0.5));
-  s = ddadd2_d2_d2_d(s,  ql * (-PI_B*0.5));
-  s = ddadd2_d2_d2_d(s, dqh * (-PI_C*0.5));
-  s = ddadd2_d2_d2_d(s,  ql * (-PI_C*0.5));
-  s = ddadd_d2_d2_d(s, (dqh + ql) * (-PI_D*0.5));
+  if (d < 15) {
```

Comment:
Why 15? Please add a comment. If it is used in many places, you might want to use a `#define`  and add the comment only in there.